app/models/assignment.rb

```diff
@@ -600,6 +600,20 @@ def summary_json(user)
       numMarked: self.get_num_marked(user.admin? ? nil : user.id) }
   end
 
+  # Generate a JSON summary of the test results associated with an assignment
+  def summary_test_result_json(user)
+    return {} unless user.admin? || user.ta?
+    {
```

Comment:
I don't think the `submission_test_results` key part is necessary: I'd like to return just the list (produced by `map`) instead.